### PR TITLE
Cleanup errors

### PR DESCRIPTION
**Pull request by pauljamescleary**
Returning an Either with ValidationError forces the caller
to handle all possible types of ValidationError even if they
are not valid for the specific function call.

Instead of a general ValidationError, return a specific validation
error for the function.

### CODE REVIEW

File: src/main/scala/io/github/pauljamescleary/petstore/validation/ValidationError.scala [hunk 1]
```diff
@@ -4,5 +4,5 @@ import io.github.pauljamescleary.petstore.model.Pet
 
 sealed trait ValidationError extends Product with Serializable
 final case class PetAlreadyExistsError(pet: Pet) extends ValidationError
-final case object PetNotFoundError extends ValidationError
-final case object OrderNotFoundError extends ValidationError
+final case class PetNotFoundError() extends ValidationError
```
Comment (tbrown1979) [inline src/main/scala/io/github/pauljamescleary/petstore/validation/ValidationError.scala]: Why the change from a `case object` over `case class`?

Comment (pauljamescleary) [reply src/main/scala/io/github/pauljamescleary/petstore/validation/ValidationError.scala]: The alternative is to define the return as `EitherT[F, PetNotFoundError.type, Pet]`

That isn't too bad, will post an update
